### DISCUSSION TITLE
Adding `max` configuration parameter for `MultilineLambdaItParameter` rule

### DISCUSSION THREAD

**Discussion by tinder-cesardiez** [detekt/detekt#5214] [Ideas]
Would it make sense to have a configuration parameter to tweak the maximum number of "it" usages on a particular scope?

so let's say we have:
```
MultilineLambdaItParameter:
    active: true
    max: 2
```
Code like the the one showcased on the [docs](https://detekt.dev/docs/next/rules/style/#multilinelambdaitparameter):
```
val digits = 1234.let { it ->
    println(it)
    listOf(it)
}
```
would now become compliant.

The reason behind the suggestion is that even though it's definitely good to have this rule in place for readability purposes, it might be a little bit limiting and noisy for our fellow devs. 

This is me thinking out loud. If you guys think it's useful, I would be happy to contribute to the repo to make it happen.

**Comment by cortinico** [comment]
Thanks for the suggestion

> The reason behind the suggestion is that even though it's definitely good to have this rule in place for readability purposes, it might be a little bit limiting and noisy for our fellow devs.

Can you elaborate on why the snippet you linked has decreased readability due to multiple `it` usages?
